pennylane/gradients/__init__.py

```diff
@@ -47,6 +47,7 @@
     param_shift_cv
     param_shift_hessian
     spsa_grad
+    hadamard_grad

```

Comment:
Do you have a feel for which of the two names is more common?

Reply:
This was my suggestion to @rmoyard in conversation the other day; we noticed that -- at least in our quantum gradient bubble -- Hadamard test was more common/informational. But it was partly informed by consistency and the fact we use `hadmard` in the name for the `metric_tensor` method

Reply:
What about `hadamard_test_grad`?

Reply:
Okay, I just noticed that https://arxiv.org/abs/1811.11184 uses the LCU moniker, and now I am wondering _why_ we call it the Hadamard test method informally

Reply:
I'm going to tag @mariaschuld here 😆

Reply:
Edit: ignore my above comments, I am completely wrong! The hadamard test method != the LCU test method

Reply:
I think the formulation that we use is based on an Hadamard test but has nothing to do with LCU, we do not use the linear combination of unitaries https://arxiv.org/abs/1811.11184  (two control gates) or https://arxiv.org/pdf/1202.5822.pdf . I think it confusing because the distinction is not clear in most paper.

Hadamard test formulation:
 1. From David's thesis I will not share the picture here, but only one control of the generator is necessary.
LCU formulation:
![Screenshot from 2023-01-24 16-08-58](https://user-images.githubusercontent.com/36766072/214415998-c91b5c34-68f7-4a03-992f-3306662ea3f9.png)
![Screenshot from 2023-01-24 16-06-27](https://user-images.githubusercontent.com/36766072/214416067-ed6bce17-c58c-439d-bb97-b77fc6cd6977.png)

Reply:
I agree with the tendency here. Another point to consider: LCU is heavily used for everything singular-value-decomposition related in (fault-tolerant) quantum algorithms, so staying away from this with the name might avoid confusion in the future :)

Reply:
Do we know how the methods compare performance-wise? Are they more-or-less similar and we only need to add one, or if there are different regimes in which Hadamard > LCU and vice versa, perhaps we can add LCU in future too.

Reply:
The LCU method has more requirements because we need to implement c-H, i.e. a controlled application of the Hamiltonian itself. For this, a decomposition of the Hamiltonian into unitaries is required if we're on quantum hardware.
The advantage one can hope for with the LCU method is that the main qubit register is not measured, allowing for error mitigation techniques like echo verification protocols.